### PR TITLE
Add npm audit github action

### PR DESCRIPTION
**Pull request by michaelbrown-stateless**
## Description

Runs NPM audit job, heavily inspired by https://blog.nishanthkp.com/docs/devsecops/sca/npm-audit/npm-audit-github/

## Focus points

Should this be called on every push?

### CODE REVIEW

File: .github/workflows/npm-audit.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: NPM Audit Scan for Vulnerabilities
+run-name: Running an Audit Scan with NPM
+on:
+  push:
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:4]: This should be `on: [push]`

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "axios": "^1.7.7",
     "axios-mock-adapter": "^1.22.0",
     "concurrently": "^8.2.2",
-    "eslint": "^9.2.0",
+    "eslint": "^8.57.0",
```
Comment (tgamber-stateless) [inline package.json:18]: ~~Why did eslint get downgraded?~~ See that it had to do with peer dependency. You can fix this by instead upgrading typescript-eslint to the latest version.

File: .github/workflows/npm-audit.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    name: NPM Audit Scan for Vulnerabilities
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:17]: We should remain consistent with our use of `yarn` as the package manager. You can add the line `cache: "yarn"`  to the `with:` section here to achieve that.

File: .github/workflows/npm-audit.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Step3 Install Dependencies
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:19]: I'd say remove the "Step3" in the name.

File: .github/workflows/npm-audit.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        run: npm install
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:20]: See note above about using `yarn`. Changes to `yarn install`

File: .github/workflows/npm-audit.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: NPM Audit Scan and Generate Report in JSON Format
+        run: npm audit --json > npm-audit-results.json
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:23]: Same note about using `yarn`. You can add `yarn` to the beginning of your run statement.

File: .github/workflows/npm-audit.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        continue-on-error: true
+
+      - name: Upload NPM Audit Scan Report to GitHub
+        uses: actions/upload-artifact@v3
```
Comment (tgamber-stateless) [inline .github/workflows/npm-audit.yml:27]: v3 getting deprecated, lets up this to v4.

**Review by tgamber-stateless — CHANGES_REQUESTED** [review]

File: .github/workflows/npm-audit.yml (hunk 1 above)
Comment (michaelbrown-stateless) [reply .github/workflows/npm-audit.yml]: Updated

File: package.json (hunk 2 above)
Comment (michaelbrown-stateless) [reply package.json]: Upgraded typescript-eslint to the latest version

File: .github/workflows/npm-audit.yml (hunk 6 above)
Comment (michaelbrown-stateless) [reply .github/workflows/npm-audit.yml]: Added

File: .github/workflows/npm-audit.yml (hunk 7 above)
Comment (michaelbrown-stateless) [reply .github/workflows/npm-audit.yml]: Made this change for vacuum as well

File: .github/workflows/lint-spec.yml [hunk 8]
```diff
@@ -1,21 +1,20 @@
...
+      - name: Generate Vacuum HTML Reports
+        run: mkdir -p reports; for FILEPATH in specs/ccapi*.json; do FILE="$(echo $FILEPATH  | cut -d '/' -f2 | cut -d '.' -f1)"; vacuum html-report "${FILEPATH}" "reports/report-${FILE}.html"; done
+        shell: bash
+      - name: Archive Vacuum HTML Reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: Vacuum HTML Reports
+          path: reports/*.html
+      - name: Lint with Vaccum
+        run: vacuum lint specs/ccapi* || true
```
Comment (daveset) [inline .github/workflows/lint-spec.yml:19]: I know this is just reformatted but I think a short comment here about why we are ignoring failures may be helpful to future reviewers of this. Also on line 18 please change `Vaccum` to `Vacuum`.

**Comment by JimTheCactus** [comment]
As a heads up, this will also run on merge. This is undesirable at the moment, and it'd be ideal if we can find a way to avoid running when this gets merged into the final repo (i.e. CI only runs on the developer's account, not CC's)

**Review by tgamber-stateless — COMMENTED** [review]
The branch-ignore: main strategy should keep workflows from running on merge for the moment. The filters may need to be expanded in the future, but as far as I can tell this is the best way to have things work as desired besides just disabling actions in the main repo. 

Anyways, just have a quick suggestion about bumping the upload-artifact version in the lint-spec file.

**Review by daveset — APPROVED** [review]
I disagree with the idea of disabling CI checks on pushes to main, but until we hear differently, that is what we are doing. Approved... grudgingly

**Review by tgamber-stateless — APPROVED** [review]
With that last change, looks ready to me.

**Pull request merged**